coderd/rbac/authz.go

```diff
@@ -11,24 +11,35 @@ import (
 
 type Authorizer interface {
 	ByRoleName(ctx context.Context, subjectID string, roleNames []string, action Action, object Object) error
+	PrepareByRoleName(ctx context.Context, subjectID string, roles []string, action Action, objectType string) (PreparedAuthorized, error)
```

Comment:
I don't see any use case for exposing the `PreparedAuthorized` to consumers of authorization --- why not just expose the Filter() method here?

Reply:
I wish I could, but you can't make generic methods on structs. The nice thing about the generic `Filter()` method is you can pass in and return the same typed slice.

If I made it a non-generic function, you'd have to take your slice of `[]Workspace`, convert it to `[]rbac.Object`, and then convert it back to `[]Workspace`. I could do this all in some `rbac.Authorize`, but it takes a lot of slice copying I prefer to avoid.

Reply:
Well, that sucks.  Go generics continue to disappoint.

Reply:
Yea, it's really dumb because I don't see why it's not possible. Like I would like to know the objection to it.